include/fmt/core.h

```diff
@@ -82,11 +82,17 @@
 #  define FMT_HAS_CPP_ATTRIBUTE(x) 0
 #endif
 
+#ifdef _MSVC_LANG
+#  define FMT_CPLUSPLUS _MSVC_LANG
+#else
+#  define FMT_CPLUSPLUS __cplusplus
+#endif
+
 #define FMT_HAS_CPP14_ATTRIBUTE(attribute) \
-  (__cplusplus >= 201402L && FMT_HAS_CPP_ATTRIBUTE(attribute))
+  (FMT_CPLUSPLUS >= 201402L && FMT_HAS_CPP_ATTRIBUTE(attribute))
 
 #define FMT_HAS_CPP17_ATTRIBUTE(attribute) \
-  (__cplusplus >= 201703L && FMT_HAS_CPP_ATTRIBUTE(attribute))
+  (FMT_CPLUSPLUS >= 201703L && FMT_HAS_CPP_ATTRIBUTE(attribute))
```

Comment:
I'm not even sure why guard against C++17 attributes in C++14 mode if compiler exposes them unconditionally. Compilers are happy to warn even in C++14 mode: https://godbolt.org/z/TrrKhq355

Reply:
```
<source>:3:3: warning: use of the 'nodiscard' attribute is a C++17 extension [-Wc++17-extensions]
[[nodiscard]] int f();
  ^
```
that's why

Reply:
uh, ok